src/cubic_loader/qlik/ods_qlik.py

```diff
@@ -380,6 +381,19 @@ def rds_fact_table_load(self) -> None:
 
         logger.log_complete()
 
+    def snapshot_reset(self) -> None:
+        """
+        Reset database tables and status files for new snapshot
+
+        Those will lose all history from history table
+        """
+
+        s3_delete_object(self.status_path)
+        self.etl_status = self.load_etl_status()
+
+        self.db.execute(drop_table(self.db_history_table))
```

Comment:
I can't tell but are these within a db transaction. Also, do we need delete permissions for the database?

Reply:
the `self.db.execute` call automatically creates it's own transaction, but each call is in its own transaction.

The application role should automatically have permission to drop these tables because the role also created them. That is generally standard in postgres.